F-FrontEnd/src/F-output-xcodeml.c

```diff
@@ -53,15 +53,58 @@ static void mark_type_desc_in_id_list(ID ids);
 char s_timestamp[CEXPR_OPTVAL_CHARLEN] = {0};
 char s_xmlIndent[CEXPR_OPTVAL_CHARLEN] = "  ";
 
-EXT_ID current_function_stack[MAX_UNIT_CTL_CONTAINS + 1];
+#define CURRENT_FUNCTION_STACK_MAX_SIZE MAX_UNIT_CTL_CONTAINS + 100
+
+static const size_t current_function_stack_max_size = CURRENT_FUNCTION_STACK_MAX_SIZE;
+EXT_ID current_function_stack[CURRENT_FUNCTION_STACK_MAX_SIZE];
 int current_function_top = 0;
-#define CRT_FUNCEP current_function_stack[current_function_top]
-#define CRT_FUNCEP_PUSH(ep)                                                    \
-    current_function_top++;                                                    \
-    current_function_stack[current_function_top] = (ep)
-#define CRT_FUNCEP_POP                                                         \
-    current_function_stack[current_function_top] = (ep);                       \
-    current_function_top--
+
+static inline EXT_ID GET_CRT_FUNCEP()
+{
+	if(current_function_top >= 0 && current_function_top < current_function_stack_max_size)
+	{
+		return current_function_stack[current_function_top];
+	}
+    else
+    {
+    	fatal("read out of bounds, current_function_stack_max_size (%d)", current_function_stack_max_size);
+    	return NULL;//Unreachable code
+    }
+}
```

Comment:
I know this is just merged but indent is completely inconsistent. `if`/`else` are not aligned. Before `return` in `if` block has more whitespaces than the one in `else` block. 

Same comment for the functions below.

Reply:
I guess github uses more than 4 spaces to display a tab, that's why it looks misaligned.

Reply:
https://github.com/omni-compiler/xcodeml-tools/pull/179